engine/src/main/java/org/camunda/bpm/engine/impl/bpmn/deployer/BpmnDeployer.java

```diff
@@ -185,11 +184,17 @@ protected void updateJobDeclarations(List<JobDeclaration<?, ?>> jobDeclarations,
 
       for (JobDeclaration<?, ?> jobDeclaration : jobDeclarations) {
         boolean jobDefinitionExists = false;
+        // find matching job definition entity
         for (JobDefinition jobDefinitionEntity : existingDefinitions) {
 
-          // <!> Assumption: there can be only one job definition per activity and type
-          if(jobDeclaration.getActivityId().equals(jobDefinitionEntity.getActivityId()) &&
-              jobDeclaration.getJobHandlerType().equals(jobDefinitionEntity.getJobType())) {
+          // activity id needs to match
+          boolean activityIdMatches = jobDeclaration.getActivityId().equals(jobDefinitionEntity.getActivityId());
+          // handler type (e.g. 'async-continuation' needs to match
+          boolean handlerTypeMatches = jobDeclaration.getJobHandlerType().equals(jobDefinitionEntity.getJobType());
+          // configuration (e.g. 'async-before', 'async-after' needs to match
+          boolean configurationMatches = jobDeclaration.getJobConfiguration().equals(jobDefinitionEntity.getJobConfiguration());
+
+          if(activityIdMatches && handlerTypeMatches && configurationMatches) {
```

Comment:
Fair point, thanks! I don't think the impact is relevant. All operations are simple getter calls. The code is called upon deployment of BPMN files and loops through all existing ProcessDefinitions. So it doesn't run often and the count of ProcessDefinitions usually does not reach high numbers.
The benefit of this approach is better readability while also making it easier to add those comments. I think they have value because it might not be clear why those checks are performed.